dotnet/samples/KernelSyntaxExamples/Example76_StronglyTypedKernelResult.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Threading.Tasks;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel;
+using System.Text.Json;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+// ReSharper disable once InconsistentNaming
+public static class Example76_StronglyTypedKernelResult
+{
+    public static async Task RunAsync()
+    {
+        Console.WriteLine("======== LLMPrompts ========");
+
+        string openAIModelId = TestConfiguration.OpenAI.ChatModelId;
+        string openAIApiKey = TestConfiguration.OpenAI.ApiKey;
+
+        if (openAIApiKey == null)
+        {
+            Console.WriteLine("OpenAI credentials not found. Skipping example.");
+            return;
+        }
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddOpenAIChatCompletion(
+                modelId: openAIModelId,
+                apiKey: openAIApiKey)
+            .Build();
+
+        var promptTestDataGeneration = "Return a JSON with an array of 3 JSON objects with the following fields: First, an id field with a random GUID, next a name field with a random company name and last a description field with a random short company description. Ensure the JSON is valid and it contains a JSON array named testcompanies with the three fields.";
+
+        // Time it
+        var sw = new Stopwatch();
+        sw.Start();
+
+        FunctionResult functionResult =
+            await kernel.InvokePromptAsync(promptTestDataGeneration);
+
+        // Stop the timer
+        sw.Stop();
+
+        var functionResultTestDataGen = new FunctionResultTestDataGen(
+            functionResult!,
+            sw.ElapsedMilliseconds);
+
+        Console.WriteLine($"Test data: {functionResultTestDataGen.Result} \n");
+        Console.WriteLine($"Milliseconds: {functionResultTestDataGen.ExecutionTimeInMilliseconds} \n");
+        Console.WriteLine($"Total Tokens: {functionResultTestDataGen.TokenCounts!.TotalTokens} \n");
+    }
+}
+
+// The structure to put the JSON result in a strongly typed object
+public class RootObject
+{
+    public List<TestCompany> TestCompanies { get; set; }
+}
+
+public class TestCompany
+{
+    public string Id { get; set; }
+    public string Name { get; set; }
+    public string Description { get; set; }
+}
+
+// The FunctionResult custom wrapper to parse the result and the tokens
+public record FunctionResultTestDataGen : FunctionResultExtended
```

Comment:
Suggest avoiding usage of `records` as they add unneeded complexity at the compiler time and there's no actual benefit of using it in the example.

Reply:
changing it in 1, 2...

Reply:
Done :)

Reply:
Hi @RogerBarreto and @markwallace-microsoft I think it is clean and ready...